source/common/http/filter_manager.cc

```diff
@@ -20,12 +20,10 @@ namespace Http {
 
 namespace {
 
-template <class T> using FilterList = std::list<std::unique_ptr<T>>;
-
 // Shared helper for recording the latest filter used.
-template <class T>
-void recordLatestDataFilter(const typename FilterList<T>::iterator current_filter,
-                            T*& latest_filter, const FilterList<T>& filters) {
+template <class Iterator, class Filter>
```

Comment:
Seems that the function's code requires a strict relationship between the types of Iterator and Filter. Can this be rewritten with a single template type?

Reply:
sgtm.